layouts/partials/header.html

```diff
@@ -79,14 +79,46 @@
                 <a href="javascript:void(0);" class="menu-item theme-switch" title="{{ T "switchTheme" }}">
                     <i class="fas fa-adjust fa-fw"></i>
                 </a>
-                <!-- <a href="javascript:void(0);" class="menu-item theme-switch" title="{{ T "switchTheme" }}">
-                    <select class="theme-select" id="theme-select-desktop" title="{{ T "switchTheme" }}">
-                        <option value="Light">Light</option>
-                        <option value="Dark">Dark</option>
-                        <option value="Black">Black</option>
-                        <option value="Auto">Auto</option>
+                <a href="javascript:void(0);" class="menu-item theme" title="{{ T "switchTheme" }}">
+                    <i class="fas fa-adjust fa-fw"></i>
+                    <select class="theme-select" id="theme-select-desktop" onchange="setTheme(this.value);" title="{{ T "switchTheme" }}">
+                        <option value="light">Light</option>
+                        <option value="dark">Dark</option>
+                        <option value="black">Black</option>
+                        <option value="auto">Auto</option>
                     </select>
-                </a> -->
+                </a>
+                <script>
+                    function getTheme() {
```

Comment:
The script needs to put in theme.js and update `this.isDark` to update the theme of some shortcodes.